view/frontend/web/js/product/button.js

```diff
@@ -5,22 +5,23 @@ define([
 
     return function (config) {
 
-        $(config.buttonId).on('mouseover',function (e) {
+        $(config.buttonId).on('mouseenter',function (e) {
 
-            var radioAndCheckbox = {};
+            var radioAndCheckboxElements = {};
             var inputElements = [];
 
-            $.each($('.product-options-wrapper [aria-required="true"]'),function () {
+            $.each($(config.productConfigAndOptionsSelector),function () {
 
+                /** skip swatch clearfix elements and hidden elements */
                 if(typeof $(this).prop("name") === "undefined" || $(this).attr("type") === "hidden"){
                     return;
                 }
 
                 if($(this).attr("type") === "radio" || $(this).attr("type") === "checkbox"){
 
-                    String($(this).attr("name")) in radioAndCheckbox ?
-                        radioAndCheckbox[$(this).attr("name")].push($(this).is(':checked'))
-                        : radioAndCheckbox[$(this).attr("name")] = [$(this).is(':checked')]
+                    String($(this).attr("name")) in radioAndCheckboxElements ?
+                        radioAndCheckboxElements[$(this).attr("name")].push($(this).is(':checked'))
+                        : radioAndCheckboxElements[$(this).attr("name")] = [$(this).is(':checked')]
```

Comment:
I meant breaking up this whole like into different parts since it's so long, not just any specific parts

Reply:
It's not long, it's compact.

Reply:
ok then maybe a normal if statement instead of ternary would be better since even though each part is only one line, it's still pretty long and stretches the one statement out to 3 lines